include/openthread/ip6.h

```diff
@@ -114,6 +114,15 @@ OTAPI otError OTCALL otIp6RemoveUnicastAddress(otInstance *aInstance, const otIp
 OTAPI const otNetifAddress *OTCALL otIp6GetUnicastAddresses(otInstance *aInstance);
 
 /**
+ * The function returns the link-local IPv6 address assigned to the Thread interface.
+ *
+ * @param[in]  aInstance A pointer to an OpenThread instance.
+ *
+ * @returns A pointer to link-local Network Interface Address, or NULL if there is no link-local address.
+ */
+const otNetifAddress *otIp6GetLinkLocalAddress(otInstance *aInstance);
```

Comment:
I was thinking of the same after submitting this. Even though, we may have one ll address, technically we can have more than one. 

`wpantund` also sort of assumes there is a single LL IPv6 address on the thread interface, so the spinel property `IPV6_LL_ADDR` is a single IPV6 encoding (not an array) and within wpantund the address is saved as a single IP address in `NCPInstanceBase::mNCPLinkLocalAddress`